src/settings.rs

```diff
@@ -1,6 +1,6 @@
 //! Application settings objects and initialization
 
-use std::collections::HashMap;
+use std::collections::BTreeMap;
```

Comment:
Why the switch from HashMap to BTreeMap (and likewise, the use of BTreeSet)?

Reply:
For the sorts of lookups we're going to be doing, (mostly alpha based, and not a lot of entries) a BTreeMap might be a bit faster than hashing the entire string to do basically the same thing. (Tempted to also do case normalization here too, to be honest.)

Reply:
Yeah, the difference should be super minimal, and we could easily swap between the two.